src/cache.cpp

```diff
@@ -69,11 +73,11 @@ Node* CacheTree::construct_node(unsigned short new_rule, size_t nrules, bool pre
  * Inserts the root of the tree, setting up the default rules.
  */
 void CacheTree::insert_root() {
-    VECTOR tmp_vec;
+    //VECTOR tmp_vec;
     size_t d0, d1;
     bool default_prediction;
     double objective;
-    make_default(&tmp_vec, nsamples_);
+    //make_default(&tmp_vec, nsamples_);
```

Comment:
If nsamples_ isn't initialized, how can you use it d1?

Reply:
It's initialized in line 31 of cache.cpp, in CacheTree's constructor